frontend/viewer/src/lib/components/ui/format/format-number.ts

```diff
@@ -0,0 +1,14 @@
+import {fromStore} from 'svelte/store';
+import {locale} from 'svelte-i18n-lingui';
+
+const stateLocale = fromStore(locale);
+
+export function formatNumber(value: number | undefined, options?: Intl.NumberFormatOptions, defaultValue = ''): string {
+  if (value === undefined) return defaultValue;
+
+  return new Intl.NumberFormat(stateLocale.current, {
```

Comment:
I wonder if we should be using: https://lingui.dev/ref/core#i18n.number instead. Though I'm not sure if it'll be setup correctly so I'm not totally sure.